packages/docprovider/src/yprovider.ts

```diff
@@ -70,69 +50,68 @@ export class WebSocketProvider
     user.changed.connect(userChanged);
   }
 
-  get renameAck(): Promise<boolean> {
-    return this._renameAck.promise;
-  }
+  get ready(): Promise<boolean> {
+    if (this._ready) {
+      return this._ready.promise;
+    } else {
+      this._ready = new PromiseDelegate();
+    }
 
-  setPath(newPath: string): void {
-    if (newPath !== this._path) {
-      this._path = newPath;
-      const encoder = encoding.createEncoder();
-      this._renameAck = new PromiseDelegate<boolean>();
-      encoding.write(encoder, 127);
-      // writing a utf8 string to the encoder
-      const escapedPath = unescape(
-        encodeURIComponent(
-          this._format + ':' + this._contentType + ':' + newPath
-        )
-      );
-      for (let i = 0; i < escapedPath.length; i++) {
-        encoding.write(
-          encoder,
-          /** @type {number} */ escapedPath.codePointAt(i)!
+    const serverSettings = ServerConnection.makeSettings();
```

Comment:
Do we need to wait for `ready` access to do this request? Can't we start it right away in the constructor as we already know the context to fasten the initialization?

Reply:
Yes, thanks, see https://github.com/jupyterlab/jupyterlab/pull/13246/commits/12ae3109abc8f5e213ea64506923ca8525815823.